source/agora/node/FullNode.d

```diff
@@ -146,14 +146,38 @@ public class FullNode : API
             400, Json("The query was incorrect"), string.init, int.init);
     }
 
-    /// The first task method, loading from disk, node discovery, etc
+    /***************************************************************************
+
+        Begins asynchronous tasks for node discovery and periodic catchup.
+
+    ***************************************************************************/
+
     public void start ()
     {
+        this.startPeriodicDiscovery();
+        this.network.startPeriodicCatchup(this.ledger);
+    }
+
+    /***************************************************************************
+
+        Starts the periodic network discovery task.
+
+    ***************************************************************************/
+
+    private void startPeriodicDiscovery ()
+    {
+        import core.time;
+
         this.taskman.runTask(
+        ()
         {
-            log.info("Doing network discovery..");
-            this.network.discover();
-            this.network.startPeriodicCatchup(this.ledger);
+            while (1)
+            {
+                scope (success)
+                    this.taskman.wait(5.seconds);
```

Comment:
Because I don't want to suspend the fiber if an exception is thrown. I want it to kill the fiber instead.

And now that I said that out loud, I realize I should probably make `discover` nothrow.

Reply:
Buuuuuuuttttttttt... Why not simply:
```D
while (true)
{
    this.network.discover();
    this.taskman.wait(5.seconds);
}
```

The wait is in a `scope (success)` in a loop with a single statement. Either the statement throws and the code is never executed, or the statement doesn't throw and the code is executed. What am I missing ?

Reply:
Ahh right, my bad. It was copied over from the other example usage where I have a `continue`. I'll fix it.